Makefile

```diff
@@ -203,117 +102,103 @@ files: all
+	mkdir -p docs/man/man1
+	./wwctl --emptyconf genconfig man docs/man/man1
+	gzip --force docs/man/man1/*.1
+	gzip --force --keep docs/man/man5/*.5
 
-man_pages: wwctl
-	@install -d man_pages
-	@./wwctl --emptyconf genconfig man man_pages 
-	@cp docs/man/man5/*.5 ./man_pages/
-	@echo -n "Compressing manpage: "
-	@cd man_pages; for i in wwctl*1 *.5; do gzip --force $$i; echo -n "$$i "; done; echo
+update_configuration: vendor $(call godeps,cmd/update_configuration/update_configuration.go)
+	go build -X 'github.com/hpcng/warewulf/internal/pkg/node.ConfigFile=./etc/nodes.conf'" \
+	    -mod vendor -tags "$(WW_GO_BUILD_TAGS)" -o update_configuration cmd/update_configuration/update_configuration.go
 
-update_configuration: vendor cmd/update_configuration/update_configuration.go
-	cd cmd/update_configuration && go build \
-	 -X 'github.com/hpcng/warewulf/internal/pkg/node.ConfigFile=./etc/nodes.conf'"\
-	 -mod vendor -tags "$(WW_GO_BUILD_TAGS)" -o ../../update_configuration
+wwapid: $(call godeps,internal/app/api/wwapid/wwapid.go)
+	go build -o ./wwapid internal/app/api/wwapid/wwapid.go
+
+wwapic: $(call godeps,internal/app/api/wwapic/wwapic.go)
+	go build -o ./wwapic  internal/app/api/wwapic/wwapic.go
+
+wwapird: $(call godeps,internal/app/api/wwapird/wwapird.go)
+	go build -o ./wwapird internal/app/api/wwapird/wwapird.go
 
+.PHONY: dist
 dist: vendor config
-	rm -rf .dist/$(WAREWULF)-$(VERSION) $(WAREWULF)-$(VERSION).tar.gz
+	rm -rf .dist/ $(WAREWULF)-$(VERSION).tar.gz
 	mkdir -p .dist/$(WAREWULF)-$(VERSION)
 	rsync -a --exclude=".*" --exclude "*~" * .dist/$(WAREWULF)-$(VERSION)/
 	cd .dist; tar -czf ../$(WAREWULF)-$(VERSION).tar.gz $(WAREWULF)-$(VERSION)
 	rm -rf .dist
 
+.PHONY: reference
 reference: wwctl
 	mkdir -p userdocs/reference
 	./wwctl --emptyconf genconfig reference userdocs/reference/
 
 latexpdf: reference
 	make -C userdocs latexpdf
 
-## wwapi generate code from protobuf. Requires protoc and protoc-grpc-gen-gateway to generate code.
-## To setup latest protoc:
-##    Download the protobuf-all-[VERSION].tar.gz from https://github.com/protocolbuffers/protobuf/releases
-##    Extract the contents and change in the directory
-##    ./configure
-##    make
-##    make check
-##    sudo make install
-##    sudo ldconfig # refresh shared library cache.
-## To setup protoc-gen-grpc-gateway, see https://github.com/grpc-ecosystem/grpc-gateway
-proto: 
+# wwapi generate code from protobuf. Requires protoc and protoc-grpc-gen-gateway to generate code.
+# To setup latest protoc:
+#    Download the protobuf-all-[VERSION].tar.gz from https://github.com/protocolbuffers/protobuf/releases
+#    Extract the contents and change in the directory
+#    ./configure
+#    make
+#    make check
+#    sudo make install
+#    sudo ldconfig # refresh shared library cache.
+# To setup protoc-gen-grpc-gateway, see https://github.com/grpc-ecosystem/grpc-gateway
+.PHONY: proto
+proto:
 	protoc -I /usr/include -I internal/pkg/api/routes/v1 -I=. \
```

Comment:
protoc needs following two binaries to run, `protoc-gen-go`  and `protoc-gen-go-grpc`. They can be installed with 
```
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
```
Please add an appropriate rule, like there is one for the linter.

Reply:
@mslacken I captured all four, in fact. Let me know what you think.